src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,161 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        head = null;
+        tail = null;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> tailNode = tail;
+        Node<T> nodeToAdd = new Node<>(value, tail, null);
+        tail = nodeToAdd;
+        if (tailNode == null) {
+            head = nodeToAdd;
+        } else {
+            tailNode.next = nodeToAdd;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        indexCheckForAddOperation(index);
+        Node<T> foundNode = findNodeByIndex(index);
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> previousNode = foundNode.prev;
+            Node<T> nodeToAdd = new Node<>(value, previousNode, foundNode);
+            if (previousNode != null) {
+                foundNode.prev.next = nodeToAdd;
+                foundNode.prev = nodeToAdd;
+            } else {
+                head = nodeToAdd;
+                nodeToAdd.next = foundNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexCheckForGetSetRemoveOperations(index);
+        return findNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        indexCheckForGetSetRemoveOperations(index);
+        Node<T> foundNode = findNodeByIndex(index);
+        T oldItem = foundNode.item;
+        foundNode.item = value;
+        return oldItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexCheckForGetSetRemoveOperations(index);
+        Node<T> nodeToRemove = findNodeByIndex(index);
+        unlinkNode(nodeToRemove);
+        size--;
+        return nodeToRemove.item;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> iteratorNode = head;
+        for (int i = 0; i < size; i++) {
+            if (iteratorNode.item == object
+                    || (iteratorNode.item != null && iteratorNode.item.equals(object))) {
+                unlinkNode(iteratorNode);
+                size--;
+                return true;
+            }
+            iteratorNode = iteratorNode.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(T item, Node<T> prev, Node<T> next) {
```

Comment:
```suggestion
        Node(T item, Node<T> prev, Node<T> next) {
```
won't be accessible outside of the file anyway